Sources/Implementation/DefaultDecisionService.swift

```diff
@@ -295,71 +288,126 @@ class DefaultDecisionService: OPTDecisionService {
+                                                                rules: rolloutRules,
+                                                                ruleIndex: index,
+                                                                user: user,
+                                                                options: options)
             reasons.merge(decisionResponse.reasons)
-            if decisionResponse.result ?? false {
-                var info = LogMessage.userMeetsConditionsForTargetingRule(userId, loggingKey)
-                logger.d(info)
-                reasons.addInfo(info)
-                
-                let decisionResponse = bucketer.bucketExperiment(config: config,
-                                                                 experiment: experiment,
-                                                                 bucketingId: bucketingId)
-                reasons.merge(decisionResponse.reasons)
-                if let variation = decisionResponse.result {
-                    info = LogMessage.userBucketedIntoTargetingRule(userId, loggingKey)
-                    logger.d(info)
-                    reasons.addInfo(info)
-                    
-                    let featureDecision = FeatureDecision(experiment: experiment, variation: variation, source: Constants.DecisionSource.rollout.rawValue)
-                    return DecisionResponse(result: featureDecision, reasons: reasons)
-                }
-                info = LogMessage.userNotBucketedIntoTargetingRule(userId, loggingKey)
-                logger.d(info)
-                reasons.addInfo(info)
-                break
-            } else {
-                let info = LogMessage.userDoesntMeetConditionsForTargetingRule(userId, loggingKey)
-                logger.d(info)
-                reasons.addInfo(info)
+            let (variation, skipToEveryoneElse) = decisionResponse.result!
+            
+            if let variation = variation {
+                let rule = rolloutRules[index]
+                let featureDecision = FeatureDecision(experiment: rule, variation: variation, source: Constants.DecisionSource.rollout.rawValue)
+                return DecisionResponse(result: featureDecision, reasons: reasons)
             }
+            
+            // the last rule is special for "Everyone Else"
+            index = skipToEveryoneElse ? (rolloutRules.count - 1) : (index + 1)
         }
         
-        // Evaluate fall back rule / last rule now
-        let experiment = rolloutRules[rolloutRules.count - 1]
+        return DecisionResponse(result: nil, reasons: reasons)
+    }
+    
+    func getVariationFromExperimentRule(config: ProjectConfig,
+                                        flagKey: String,
+                                        rule: Experiment,
+                                        user: OptimizelyUserContext,
+                                        options: [OptimizelyDecideOption]? = nil) -> DecisionResponse<Variation> {
```

Comment:
what about adding `IgnoreForcedVariation` in `OptimizelyDecideOption` ?

Reply:
An interesting idea. I also see a usage case for this option, but not convinced enough to add as a decide option. We can discuss it.